tutorials/data_persistence/c++11/temperature_subscriber.cxx

```diff
@@ -0,0 +1,48 @@
+//
+// (c) 2024 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+//
+//  RTI grants Licensee a license to use, modify, compile, and create derivative
+//  works of the Software.  Licensee has the right to distribute object form
+//  only for use with RTI products.  The Software is provided "as is", with no
+//  warranty of any type, including any warranty for fitness for any purpose.
+//  RTI is under no obligation to maintain or support the Software.  RTI shall
+//  not be liable for any incidental or consequential damages arising out of the
+//  use or inability to use the software.
+//
+
+#include <iostream>
+#include <thread>
+
+#include "rti/rti.hpp"
+#include "rti/sub/SampleProcessor.hpp"
+#include "temperature.hpp"
+
+int main(int argc, char **argv)
+{
+    dds::domain::DomainParticipant participant(0);
+    dds::topic::Topic<Temperature> topic(participant, "Temperature");
+    dds::sub::DataReader<Temperature> reader(topic);
+
+    rti::sub::SampleProcessor sample_processor;
+    sample_processor.attach_reader(
+            reader,
+            [](const rti::sub::LoanedSample<Temperature>& sample)
+            {
+                if (!sample.info().valid()) {
+                    // ignore samples with only meta-data
+                    return;
+                }
+
+                uint64_t timestamp =
+                        sample.info().source_timestamp().to_millisecs();
+
+                std::cout << sample.data().sensor_name() << ": "
+                        << sample.data().degrees() << " degrees ("
+                        << std::fixed << std::setprecision(2)
```

Comment:
setprecision(2) will change how cout works, and affect the degrees as well, which will have the effect that every time we print the degrees except the first one will have two decimal numbers. I think we can set `std::fixed << std::setprecision(2)` one line up.